src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1428,24 +1428,7 @@ private void UpdateChildResourceBindings(ResourceUpdateReason updateReason)
 		private void InnerUpdateChildResourceBindings(ResourceUpdateReason updateReason)
 		{
 			_isUpdatingChildResourceBindings = true;
-			foreach (var child in GetChildrenDependencyObjects())
-			{
-				if (!(child is IFrameworkElement) && child is IDependencyObjectStoreProvider storeProvider)
-				{
-					storeProvider.Store.UpdateResourceBindings(updateReason);
-				}
-			}
-		}
 
-		/// <summary>
-		/// Returns all discoverable child dependency objects.
-		/// </summary>
-		/// <remarks>
-		/// This method is potentially slow and should only be used where performance isn't a concern (eg updating resource bindings
-		/// when the app theme changes).
-		/// </remarks>
-		private IEnumerable<DependencyObject> GetChildrenDependencyObjects()
```

Comment:
This iterator was allocating too much when profiling runtime tests. The idea here is that we don't want to allocate, just iterate and execute the code we want directly.